kcp/login.sh

```diff
@@ -0,0 +1,236 @@
+        Display this message.
+
+Example:
+    ${0##*/} --all
+" >&2
+}
+
+
+
+parse_args(){
+  KCP_ENV="kcp-unstable"
+
+  local args
+  args="$(getopt -o dhk:w: -l "debug,help,kcp,workspace" -n "$0" -- "$@")"
+  eval set -- "$args";
+  while true; do
+    case $1 in
+      -k|--kcp)
+        shift
+        case $1 in
+          stable|unstable)
+            KCP_ENV="kcp-$1"
+            ;;
+          *)
+            echo "Unsupported environment: $1" >&2
+            echo "" >&2
+            usage
+            exit 1
+            ;;
+        esac
+        ;;
+      -w|--workspace)
+        shift
+        KCP_WORKSPACE="$1"
+        ;;
+      -d|--debug)
+        set -x
+        ;;
+      -h|--help)
+        usage
+        exit 0
+        ;;
+      --)
+        # End of arguments
+        break
+      ;;
+      *)
+        echo "Unknown argument: $1"
+        usage
+        exit 1
+        ;;
+    esac
+    shift
+  done
+}
+
+
+
+generate_kcp_kubeconfig(){
+  echo 'apiVersion: v1
+clusters:
+- cluster:
+    server: https://kcp-stable.apps.kcp-internal-1.52p3.p1.openshiftapps.com/clusters/root:rh-sso-15826143
+  name: kcp-stable
+- cluster:
+    server: https://kcp-unstable.apps.kcp-internal-1.52p3.p1.openshiftapps.com/clusters/root:rh-sso-15826143
+  name: kcp-unstable
+contexts:
+- context:
+    cluster: kcp-stable
+    namespace: default
+    user: oidc
+  name: kcp-stable
+- context:
+    cluster: kcp-unstable
+    namespace: default
+    user: oidc
+  name: kcp-unstable
+current-context: kcp-stable
+kind: Config
+preferences: {}
+users:
+- name: oidc
+  user:
+    exec:
+      apiVersion: client.authentication.k8s.io/v1beta1
+      args:
+      - oidc-login
+      - get-token
+      - --oidc-issuer-url=https://sso.redhat.com/auth/realms/redhat-external
+      - --oidc-client-id=rhoas-cli-prod
+      command: kubectl
+      env: null
+      interactiveMode: IfAvailable
+      provideClusterInfo: false
+' > "$KUBECONFIG_KCP"
+}
+
+
+
+login_kcp(){
+  echo -n " - KCP ($KCP_ENV): "
+  # Login in KCP
+  export KUBECONFIG="$KUBECONFIG_KCP"
+
+  if [ ! -e "$KUBECONFIG" ]; then
+    generate_kcp_kubeconfig
+  fi
+  oc config use-context $KCP_ENV >/dev/null
+  while ! oc api-resources >/dev/null ; do
+    /opt/google/chrome/chrome https://www.redhat.com/wapps/ugc/sso/logout >/dev/null 2>&1
+    find "$HOME/.kube/cache/oidc-login" -type f -delete
+    generate_kcp_kubeconfig
+    sleep 3
+  done
+}
+
+
+
+login_workspace() {
+  echo -n " - KCP Workspace ($KCP_WORKSPACE): "
+  # Create and use the workspace
+  export KUBECONFIG="$KUBECONFIG_KCP"
+  KCP_WORKSPACE="${KCP_WORKSPACE:-default}"
+  while ! kubectl kcp workspace use "$KCP_WORKSPACE" >/dev/null; do
+    kubectl kcp workspace create "$KCP_WORKSPACE"
+  done
+
+  # Fix kubeconfig
+  # Check if this workaround is still needed at the end of Q2-2022
+  sed -i -e '/ \+certificate-authority-data:.*/d' -e 's%\(\.com\):[0-9]\+%\1%' -e 's%\(clusters\)/\(rh-sso\)%\1/root:\2%' $KUBECONFIG
```

Comment:
I think this is already fixed. I don't see this issue from a couple days after upgrading the plugin. Please verify.

Reply:
Correct. Removed.